File: ee25btech11001/MATGEO/5.7.14/codes/code.py
import numpy as np
import matplotlib.pyplot as plt

# Given matrix A
a11, a12, a21, a22 = -3, 6, -2, 4

# Compute trace and determinant
trace = a11 + a22
det = a11*a22 - a12*a21

# Polynomial coefficients
coeffs = [1, -trace, det]
print("Characteristic Polynomial Coefficients:", coeffs)

# Define polynomial
lam = np.linspace(-10, 10, 400)
poly_vals = coeffs[0]*lam**2 + coeffs[1]*lam + coeffs[2]

# Plot
plt.axhline(0, color='black', linewidth=0.8)
plt.plot(lam, poly_vals, label="Characteristic Polynomial")
plt.xlabel("λ")
plt.ylabel("p(λ)")
plt.legend()
plt.grid(True)
plt.show()

File: ee25btech11001/MATGEO/5.7.14/codes/code.c
#include <stdio.h>

// Function to compute characteristic polynomial coefficients of a 2x2 matrix
// Input: a11, a12, a21, a22
// Output: coeffs[0] = 1 (λ^2), coeffs[1] = -trace(A), coeffs[2] = det(A)
void char_poly(double a11, double a12, double a21, double a22, double* coeffs) {
    double trace = a11 + a22;
    double det = a11 * a22 - a12 * a21;

    coeffs[0] = 1.0;       // λ^2 coefficient
    coeffs[1] = -trace;    // λ coefficient
    coeffs[2] = det;       // constant term
}

